integration-tests/ws-security-server/src/test/java/io/quarkiverse/cxf/it/ws/securitypolicy/server/CxfWssSecurityPolicyServerTest.java

```diff
@@ -191,9 +196,15 @@ void wsdl() {
     }
 
     WssSecurityPolicyHelloService getPlainClient() {
-        return QuarkusCxfClientTestUtil.getClient(
+        WssSecurityPolicyHelloService client = QuarkusCxfClientTestUtil.getClient(
                 "https://quarkiverse.github.io/quarkiverse-docs/quarkus-cxf/ws-securitypolicy",
                 WssSecurityPolicyHelloService.class,
                 "/soap/security-policy-hello");
+
+        //programmatic registration of customizedAlgorithmSuite
+        final Config config = ConfigProvider.getConfig();
+        new CustomAlgSuiteLoader(((JaxWsClientProxy) Proxy.getInvocationHandler(client)).getClient().getBus(), config);
```

Comment:
@ffang The same issue as is in ws-security-server tests. I have to register custom algorithm in this test. My code does not seem like a user preferred solution. Is it possible to achieve the same via some nicer configuration?

Reply:
Hi @JiriOndrusek ,

I think this is OK for now.

Freeman

Reply:
Hi @ffang 

I debugged this part of the test and there is a difference between natie and JVM mode. I'm setting the customAlgSuite via  `quarkus.cxf.customizedAlgorithmSuite.enabled = true`
- in JVM - bus created via QuarkusBusFactory is customized and the plain client get the same bus -> works without this line
- in native - bus created via QuarkusBusFactory is customized, but plain client is creating its own bus, which is not affected by the QuarkusBusFactory. My understanding is that the client is created by in the test (non-native) part of executions. Therefore I had to customize the client's bus to make scenario work.

I think that in JVM code works without this configuration, because both plain client (test resource) and app part (from the `main` source folder) share the customAlgSuite registered into a static variable

Reply:
@JiriOndrusek, this kind of differences is caused by the fact that in native mode, the tests run in a JVM, but the application under test is a fully different (native) process. Hence of you register something on the Bus in the application, it is not visible for the JVM running the tests. 

In JVM mode, the tests and the application under test are a part of the same JVM.

So you either have to configure the stuff twice, or you have to move the SOAP client to the application and wrap it in REST. I think the second option is better, because it allows to cover also the client in native.

Reply:
I agree with the cause of the issue. I'll move the client creation into the application, where the customization configuration should take care of it. I'll fix it in the same way on the other places to have a consistent approach in all tests.